modules/post/windows/gather/forensics/memorydump.rb

```diff
@@ -0,0 +1,230 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+#
+# This Module has an embred Belkasoft RAM Capturer version of 2018-08
+# Available at https://belkasoft.com/ram-capturer
+##
+
+#
+# Forensic memory dump using Belkasoft RAM Capturer
+#
+# Helvio Junior (M4v3r1cK) m4v3r1ck.hjr [at] gmail.com
+#    https://www.helviojunior.com.br
+#
+# Other modules repository: https://github.com/helviojunior/metasploit_modules
+#
+# Standard Library
+#
+require 'tmpdir'
+
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+  include Msf::Post::Windows::UserProfiles
+
+  def initialize(info={})
+    super( update_info( info,
+      'Name'          => 'Windows Gather Memory Dump Imaging',
+      'Description'   => %q{This module will perform a memory dump of remote machine},
+      'License'       => MSF_LICENSE,
+      'Platform'      => ['win'],
+      'SessionTypes'  => ['meterpreter'],
+      'Author'        => ['Helvio Junior (M4v3r1cK) <m4v3r1ck.hjr[at]gmail.com>']
+    ))
+    register_options(
+      [
+        OptString.new('RAMCAPTURE_PATH',[true,'Path of Belkasoft RAM Capturer',''])
+      ])
+  end
+
+  def run
+    vprint_status("Determining session platform, type and privileges")
+    case session.platform
```

Comment:
This code block doesn't make much sense as a `case` block.

Also, `fail_with` is preferred over `return`.

Also, `@platform` can be removed, as it is only used in the `got_root` method, which can also be removed as per comments below.

```ruby
    unless session.platform == 'windows'
      fail_with Failure::BadConfig, "Unsupported platform: #{session.platform}"
    end

    unless session.type == 'meterpreter'
      fail_with Failure::BadConfig, "Only meterpreter sessions are supported on Windows hosts"
    end
```

Reply:
@OJ Are you sure? I'm pretty sure Metasploit will throw a warning if the `SessionType` doesn't match, but continue with execution anyway. Could be wrong. See #6316